src/api-v1.js

```diff
@@ -740,26 +740,64 @@ api.declare({
   route: '/state/:workerType',
   name: 'state',
   title: 'Get AWS State for a worker type',
-  scopes: [['aws-provisioner:view-worker-type:<workerType>']],
+  output: 'state-response.json#',
   stability:  base.API.stability.stable,
   description: [
-    'Return the state of a given workertype as stored by the provisioner. ',
-    'This state is stored as three lists: 1 for all instances, 1 for requests',
-    'which show in the ec2 api and 1 list for those only tracked internally',
-    'in the provisioner.',
+    'Return the dynamic state of a given workerType, giving a large-scale view',
+    'of its allocated and available capacity.  This information is available both',
+    'by instance count and by capacity value.',
   ].join('\n'),
 }, async function (req, res) {
-  let workerType = req.params.workerType;
+  let workerType, workerState;
   try {
-    let workerState = await this.WorkerState.load({workerType: workerType});
-    res.reply(workerState._properties);
+    // need some data from the WorkerState, and some from the WorkerType
+    // TODO: parallelize this with Promise.all(..)
+    workerType = await this.WorkerType.load({workerType: req.params.workerType});
+    workerState = await this.WorkerState.load({workerType: req.params.workerType});
   } catch (err) {
     if (err.code === 'ResourceNotFound') {
       res.status(404).json({
-        message: workerType + ' does not have any state information',
+        message: req.params.workerType + ' does not have any state information',
       }).end();
+      return;
     }
+    throw err;
   }
+
+  let rv = {
+    running: {
+      instances: 0,
+      capacity: 0,
+    },
+    pending: {
+      instances: 0,
+      capacity: 0,
+    },
+    spotReq: {
+      instances: 0,
+      capacity: 0,
+    },
+    maxCapacity: workerType.maxCapacity,
+  };
+
+  let capacities = {};
+  workerType.instanceTypes.forEach(instanceType => {
+    capacities[instanceType.instanceType] = instanceType.capacity;
+  });
+
+  workerState.instances.forEach(instance => {
+    // count every AWS state other than 'running' as 'pending'
```

Comment:
we should only ever have running and pending.  Let's ignore instances that aren't in either of these states.